src/providers/AppContext/AppContext.provider.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import { YOUTUBE_KEY } from '../../utils/constants';
+
+const AppContext = React.createContext({});
+
+export const AppContextProvider = ({ children }) => {
+  const [selectedVideoId, setSelectedVideoId] = useState('');
+  const [selectedVideoTitle, setSelectedVideoTitle] = useState('');
+  const [selectedVideoDescription, setSelectedVideoDescription] = useState('');
+  const [youTubeKey] = useState(YOUTUBE_KEY);
```

Comment:
In this case the API key is a constant, so you don't need to create a state handler for this value, you can pass it as it is